src/services/kratos/auth-email-no-password.ts

```diff
@@ -1,18 +1,68 @@
 import { getKratosPasswords } from "@config"
 
-import { PhoneCodeInvalidError } from "@domain/phone-provider"
-
 import { wrapAsyncFunctionsToRunInSpan } from "@services/tracing"
 
 import { isAxiosError } from "axios"
 
-import { kratosPublic } from "./private"
-import { UnknownKratosError } from "./errors"
+import {
+  EmailCodeInvalidError,
+  EmailNotVerifiedError,
+  EmailValidationSubmittedTooOftenError,
+  LikelyUserAlreadyExistError,
+} from "@domain/authentication/errors"
+
+import { UpdateIdentityBody } from "@ory/client"
+
+import { checkedToEmailAddress } from "@domain/users"
+
+import knex from "knex"
+
+import { IncompatibleSchemaUpgradeError, KratosError, UnknownKratosError } from "./errors"
+import {
+  kratosAdmin,
+  kratosPublic,
+  toDomainIdentityEmail,
+  toDomainIdentityEmailPhone,
+  toDomainIdentityPhone,
+} from "./private"
+import { SchemaIdType } from "./schema"
+
+const getKratosKnex = () =>
+  knex({
```

Comment:
Why not use a simpler client like https://node-postgres.com?

Reply:
no specific reason, but mainly because we use knex in other repos (ie: https://github.com/GaloyMoney/price/blob/3cec863ef696e331d2e4313f83a32e6ede666770/history/src/config/database.ts#L1)